cylc/flow/scheduler.py

```diff
@@ -1878,3 +1870,32 @@ def _get_events_conf(self, key, default=None):
         """Return a named [cylc][[events]] configuration."""
         return self.suite_event_handler.get_events_conf(
             self.config, key, default)
+
+    def process_cylc_stop_point(self):
```

Comment:
I would have thought a unit test for this method would be better suited than a functional test

Reply:
I considered it, but because it's a method of the class it'd require some in un-fun mocking. However it'd be nicer, and I like the idea of working out how.